File: TypeScript/Linked-Lists/DoubleListNode.ts
/**
 * Representing a Node class in a Doubly Linked List
 * @template T the type of data stored in the node.
 * 
 * @property data: the data stored in the node
 * @property previous : A reference(pointer) to the previous element
 * @property next: A reference(pointer) to the next element, by default it is null.
 */
export class DoubleListNode<T> {
    constructor(
        public data: T,
        public previous?: DoubleListNode<T>,
        public next?: DoubleListNode<T>
    ){};
}

File: Java/LinkedLists/DoubleListNode.java
package LinkedLists;

public class DoubleListNode<T> implements IListNode<T>{

    T data;
    DoubleListNode<T> next;
    DoubleListNode<T> previous;

    public DoubleListNode(T data){
        this.data = data;
    }

    public DoubleListNode(T data, DoubleListNode<T> next) {
        this.data = data;
        this.next = next;
    }

    public DoubleListNode (T data, DoubleListNode<T> next, DoubleListNode<T> previous) {
        this.data = data;
        this.next = next;
        this.previous = previous;
    }
}
